src/relayFeeCalculator/relayFeeCalculator.ts

```diff
@@ -11,11 +11,19 @@ import {
   percent,
   MAX_BIG_INT,
   isDefined,
+  getTokenInformationFromAddress,
 } from "../utils";
+import { DEFAULT_SIMULATED_RELAYER_ADDRESS } from "../constants";
+import { Deposit } from "../interfaces";
 
 // This needs to be implemented for every chain and passed into RelayFeeCalculator
 export interface QueryInterface {
-  getGasCosts: () => Promise<BigNumberish>;
+  getGasCosts: (
+    deposit: Deposit,
+    amountToRelay: BigNumberish,
+    relayerAddress: string,
```

Comment:
qq: this parameter is optional in baseQuery, but mandatory here - is that intentional?

Reply:
It is not optional in BaseQuery. There is a defined default so that the value is never `undefined`. We can't really define this in an interface, hence we need to write `relayerAddress: string` rather than `relayerAddress?: string`

Reply:
Sorry, clumsy phrasing from my side. By `optional` I meant from the caller's perspective. But I see the reason that it needs to be a required field here :+1:

Reply:
Ah, I see what you mean. 

We had the ability in `BaseQuery` to provide the default relayer. However, since this is an interface that exists outside of BaseQuery, there is a case where we implement this interface but don't have a default in which case the caller would need to supply it every time.